include/revng/Pypeline/Helpers/Python/Utils.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <vector>
+
+#include "nanobind/nanobind.h"
+
+#include "llvm/ADT/StringRef.h"
+
+#include "revng/Pypeline/Common.h"
+#include "revng/Pypeline/ObjectID.h"
+
+namespace revng::pypeline::helpers::python {
+
+inline Request convertRequests(nanobind::list &List) {
+  Request Result;
+  for (auto It1 = List.begin(); It1 != List.end(); ++It1) {
+    nanobind::list ListInner = nanobind::cast<nanobind::list>(*It1);
+    std::vector<const ObjectID *> Chunk;
+    for (auto It2 = ListInner.begin(); It2 != ListInner.end(); ++It2) {
+      Chunk.push_back(nanobind::cast<ObjectID *>(*It2));
+    }
+    Result.push_back(Chunk);
+  }
+  return Result;
+}
+
+class ManagedPyBuffer {
+private:
+  Py_buffer Buffer;
+
+public:
+  ManagedPyBuffer() { Buffer.obj = NULL; }
```

Comment:
Can we initialize `Buffer` in the constructor and prevent any further changes to it (i.e., have the rest of the API be `const`)?

Reply:
The main pain-point is that, calling `PyObject_GetBuffer` from the constructor is not trivially possible, it can fail and the caller needs to have the result code. It can be done with a factory method, but IMHO it's more effort than needed the single callsite where it's used.

Reply:
If a class takes care of handling the release of a resource, it also make sense that it takes care of acquiring it. A factory method and no public constructor looks quite reasonable to me.

Reply:
Implemented